service/src/test/java/com/kodemon/service/test/facade/FightFacadeTest.java

```diff
@@ -0,0 +1,301 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Oliver Roch
+ */
+
+@ContextConfiguration(classes = ServiceConfig.class)
+public class FightFacadeTest extends AbstractTestNGSpringContextTests {
+
+    private BeanMappingService beanMappingService;
+    private TrainerFightService trainerFightService;
+    private PokemonFightService pokemonFightService;
+    private PokemonService pokemonService;
+    private TrainerService trainerService;
+    private BadgeService badgeService;
+    private TimeService timeService;
+    private FightFacade fightFacade;
+
+    private Trainer challenger;
+    private Trainer defender;
+    private UserDTO challengerDTO;
+    private UserDTO defenderDTO;
+    private Gym targetGym;
+    private GymDTO targetGymDTO;
+    private Pokemon pikachu;
+    private Pokemon onix;
+    private PokemonDTO pikachuDTO;
+    private PokemonDTO onixDTO;
+    private FightDTO fight1;
+    private FightDTO fight2;
+    private TrainerFight fight1e;
+    private TrainerFight fight2e;
+    private List<TrainerFight> trainerFights;
+    private List<FightDTO> fights;
+
+    @BeforeClass
+    public void prepare() {
+        beanMappingService = mock(BeanMappingService.class);
+        trainerService = mock(TrainerService.class);
+        trainerFightService = mock(TrainerFightService.class);
+        pokemonFightService = mock(PokemonFightService.class);
+        pokemonService = mock(PokemonService.class);
+        badgeService = mock(BadgeService.class);
+        timeService = mock(TimeService.class);
+        fightFacade = new FightFacadeImpl(beanMappingService, trainerFightService, pokemonFightService, pokemonService,
+                trainerService, badgeService, timeService);
+
+        prepareTrainers();
+        prepareGym();
+        prepareFights();
+    }
+
+    @Test
+    public void fightForBadgeTest() {
+        Date today = new Calendar.Builder().setDate(2015, 4, 1).build().getTime();
+
+        when(trainerFightService.wasFightForBadgeSuccessful(challenger, defender)).thenReturn(true);
+        Badge badge = new Badge();
+        badge.setGym(targetGym);
+        when(badgeService.createBadgeOfGym(targetGym)).thenReturn(badge);
+        when(timeService.currentDate()).thenReturn(today);
+        fightFacade.fightForBadge(challengerDTO, targetGymDTO);
+
+        FightDTO newFightDTO = new FightDTO();
+        newFightDTO.setTargetGym(targetGymDTO);
+        newFightDTO.setChallenger(challengerDTO);
+        newFightDTO.setFightTime(today);
+        newFightDTO.setWasChallengerSuccessful(true);
+
+        TrainerFight newFight = new TrainerFight();
+        newFight.setTargetGym(targetGym);
+        newFight.setChallenger(challenger);
+        newFight.setFightTime(today);
+        newFight.setWasChallengerSuccessful(true);
+
+        when(beanMappingService.mapTo(newFightDTO, TrainerFight.class)).thenReturn(newFight);
```

Comment:
Oh, I see what you did there. So you just reproduced how the instance is created in the actual code. That's the way also :+1: